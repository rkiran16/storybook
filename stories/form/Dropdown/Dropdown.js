import { html} from "lit-html";
import Choices from "choices.js";

window.addEventListener('DOMContentLoaded', (event) => {
    new Choices('#standardDropDown', { searchEnabled: false});
});


export const Dropdown = ({error= false}) => {
   return html`
    <div class=${`ch-drop-down ${error ? 'ch-drop-down--error' : ''}`}>
        <label for="standardDropDown">Label</label>
        <select name="standardDropDown" id="standardDropDown" aria-label="Standard DropDown">
          <option value="">Placeholder Text</option>
          <option>Option 1</option>
          <option>Option 2</option>
          <option>Option 3</option>
          <option>Option 4</option>
          <option>Option 5</option>
          <option>Option 6</option>
        </select>
        <span class="error-text">
           <span class="material-icons md-16 me-2">error_outline</span>
           Error message here
        </span>
    </div>
   `
}

export const DisabledDropdown = ({error= false}) => {
  return html`
   <div class=${`ch-drop-down ${error ? 'ch-drop-down--error' : ''}`}>
       <label for="standardDropDown">Label</label>
       <select name="standardDropDown" id="standardDropDown" disabled aria-label="Standard DropDown">
         <option value="">Placeholder Text</option>
         <option>Option 1</option>
         <option>Option 2</option>
         <option>Option 3</option>
         <option>Option 4</option>
         <option>Option 5</option>
         <option>Option 6</option>
       </select>
       <span class="error-text">
          <span class="material-icons md-16 me-2">error_outline</span>
          Error message here
       </span>
   </div>
  `
}