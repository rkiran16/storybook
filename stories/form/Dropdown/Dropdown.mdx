import { Canvas, Meta, Story } from "@storybook/addon-docs";

# Drop Down

<p>
  Dropdown component is built using Choices.js library. Choices.js is a
  lightweight, configurable select box/text input plugin. Similar to Select2 and
  Selectize but without the jQuery dependency.
</p>

<p>
  For all config options, visit the{" "}
  <a href="https://github.com/Choices-js/Choices">GitHub repo</a>.
</p>

## Setup

<Canvas isColumn>
  <code>
    <div>const element = document.querySelector('.js-choice');</div>
    <div>const choices = new Choices(element);</div>
  </code>
</Canvas>

## Input Types

Choices works with the following input types, referenced in the documentation as noted.

<table class="table table-bordered">
  <thead>
    <tr>
      <th scope="col">HTML Element</th>
      <th scope="col">Documentation "Input Type"</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>input text</td>
      <td>text</td>
    </tr>
    <tr>
      <td>
        <code>select</code>
      </td>
      <td>select-one</td>
    </tr>
    <tr>
      <td>
        <code>select multiple</code>
      </td>
      <td>select-multiple</td>
    </tr>
  </tbody>
</table>
