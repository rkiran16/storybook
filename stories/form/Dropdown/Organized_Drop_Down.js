
import { html} from "lit-html";
import Choices from "choices.js";


window.addEventListener('DOMContentLoaded', (event) => {
  new Choices('#organizedDropDown', { searchEnabled: false});
});

export const OrganizedDropdown = ({error}) => {
  return html`
    <div class=${`ch-drop-down ${error ? 'ch-drop-down--error' : ''}`}>
        <label for="organizedDropDown">Label</label>
        <select name="organizedDropDown" id="organizedDropDown"  aria-label="organized DropDown">
          <option value="">Placeholder Text</option>
          <optgroup label="Swedish Cars">
            <option value="volvo">Volvo</option>
            <option value="saab">Saab</option>
          </optgroup>
          <optgroup label="German Cars">
            <option value="mercedes">Mercedes</option>
            <option value="audi">Audi</option>
          </optgroup>
        </select>
        <span class="error-text">
          <span class="material-icons md-16 me-2">error_outline</span>
          Error message here
        </span>
    </div>
  `;
}