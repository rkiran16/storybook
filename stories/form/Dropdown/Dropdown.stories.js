import { Dropdown , DisabledDropdown } from './Dropdown';
import { html} from "lit-html";
import { OrganizedDropdown } from "./Organized_Drop_Down";
import docs from "./Dropdown.mdx";

export default {
  title: "Forms/Dropdown",
  parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62aa24120581cc1a89ee0590",
    docs: {
      page: docs
    }
  },
  argTypes: {
    error: {
      name: "Error state",
      type: { name: 'boolean', required: false},
      description: "Use when the dropdown has an error",
      defaultValue: false,
      table: {
         type: { summary: 'boolean' },
         defaultValue: { summary: false },
      },
      control: { type: 'boolean'}
    }
  }
}

const DropdownTemplate = (args) => html`${Dropdown(args)}`;
const DisabledDropdownTemplate = args => html`${DisabledDropdown(args)}`;

const OrganizedTemplate = (args) => html`${OrganizedDropdown(args)}`;

export const Standard = DropdownTemplate.bind({});
Standard.args = {error: false};

export const Disabled = DisabledDropdownTemplate.bind({});
Disabled.args = {error: false};

export const Organized_Dropdown = OrganizedTemplate.bind({});
Organized_Dropdown.args = {error:false}
