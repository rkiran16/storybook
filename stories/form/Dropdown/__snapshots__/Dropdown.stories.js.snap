// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Forms/Dropdown Disabled 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
   <div class=",
        ">
       <label for=\\"standardDropDown\\">Label</label>
       <select name=\\"standardDropDown\\" id=\\"standardDropDown\\" disabled aria-label=\\"Standard DropDown\\">
         <option value=\\"\\">Placeholder Text</option>
         <option>Option 1</option>
         <option>Option 2</option>
         <option>Option 3</option>
         <option>Option 4</option>
         <option>Option 5</option>
         <option>Option 6</option>
       </select>
       <span class=\\"error-text\\">
          <span class=\\"material-icons md-16 me-2\\">error_outline</span>
          Error message here
       </span>
   </div>
  ",
      ],
      "values": Array [
        "ch-drop-down ",
      ],
    },
  ],
}
`;

exports[`Storyshots Forms/Dropdown Organized Dropdown 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=",
        ">
        <label for=\\"organizedDropDown\\">Label</label>
        <select name=\\"organizedDropDown\\" id=\\"organizedDropDown\\"  aria-label=\\"organized DropDown\\">
          <option value=\\"\\">Placeholder Text</option>
          <optgroup label=\\"Swedish Cars\\">
            <option value=\\"volvo\\">Volvo</option>
            <option value=\\"saab\\">Saab</option>
          </optgroup>
          <optgroup label=\\"German Cars\\">
            <option value=\\"mercedes\\">Mercedes</option>
            <option value=\\"audi\\">Audi</option>
          </optgroup>
        </select>
        <span class=\\"error-text\\">
          <span class=\\"material-icons md-16 me-2\\">error_outline</span>
          Error message here
        </span>
    </div>
  ",
      ],
      "values": Array [
        "ch-drop-down ",
      ],
    },
  ],
}
`;

exports[`Storyshots Forms/Dropdown Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
    <div class=",
        ">
        <label for=\\"standardDropDown\\">Label</label>
        <select name=\\"standardDropDown\\" id=\\"standardDropDown\\" aria-label=\\"Standard DropDown\\">
          <option value=\\"\\">Placeholder Text</option>
          <option>Option 1</option>
          <option>Option 2</option>
          <option>Option 3</option>
          <option>Option 4</option>
          <option>Option 5</option>
          <option>Option 6</option>
        </select>
        <span class=\\"error-text\\">
           <span class=\\"material-icons md-16 me-2\\">error_outline</span>
           Error message here
        </span>
    </div>
   ",
      ],
      "values": Array [
        "ch-drop-down ",
      ],
    },
  ],
}
`;
